### ISSUE TITLE
Connected Texture Issue

### ISSUE THREAD

**Issue by FlamenWolf** [octarine-noise/BetterFoliage#335]
For some reason, some connected textures from resource packs, like dirt, don't actually connect on the forge version. Removing the mod fixes the issue so it's definitely the mod.

**Comment by octarine-noise** [comment]
If you mean Optifine's connected texture feature, that is not supported. Not sure it even will be, as working with Optifine is pure cancer. Compatibility breaks regularly, so I'm only keeping the support for absolute necessities (i.e. shaders).

If you mean something else, please elaborate.

**Comment by FlamenWolf** [comment]
Yeah that's what I mean. I'm using the BetterVanillaBuilding resource pack. I understand if it's a pain to do. Is there any chance there's just some settings I need to disable to get them both working? I figured just turning off the connected textures part of the mod would do it but it didn't. If not that's fine. I'll still use the mod because what it adds makes Minecraft look so much better.

**Comment by Answordy** [comment]
It isn't just connected textures (ctm and all of its methods), but also custom tetxures, like custom colors and such. Optifine even breaks the Soul Sand particle. Just pointing it out, it's kinda excpected if optifine isn't supported. It is indeed annoying to work with.

 " Is there any chance there's just some settings I need to disable to get them both working?" - It would be really good though. But that would be almost the same as making them compatible Isn't it? Because even if you turn off everything in the BetterFoilage settings Optifine ctm still breaks... suggesting the problem has deeper roots. But I have no clue about modding. 
 We can only hope there will be at least some kind of, minimal/situational compatibility in the feature.

**Comment by nevis2us** [comment]
Disabling connected grass textures in the mod should probably restore default OptiFine behavior but it disables connected grass completely. Using BF connected grass leads to unpleasant checkerboard patterns on hill slopes and a number of minor glitches like dirt path textures not connecting when a road goes up one block. IMHO you should probably leave connected textures to OptiFine or at least provide a way to select either OptiFine or BF connected grass textures if at all possible. Thank you.

Screenshots:

https://ipros24.ru/2021-06-12_12-39-03/
https://ipros24.ru/2021-06-12_12-38-05/
https://ipros24.ru/2021-06-12_12-35-56/
https://ipros24.ru/2021-06-11_22-20-23/
